### PR TITLE
Update CHANGELOG for 1.0.0 release

### PR DESCRIPTION
**Pull request by kihneman**
Fixes https://github.com/enthought/okonomiyaki/issues/320

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,3 +1,16 @@
+2019-03-15     1.0.0
+--------------------
+
+Enhancements:
+
+* Add ability to parse package metadata version 2.1 (#312)
```
Comment (itziakos) [inline CHANGELOG]: `package`  -> `setuptools eggs and wheels with`

File: CHANGELOG [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+
+bug fixes:
```
Comment (itziakos) [inline CHANGELOG]: These items look that should be part of the 0.17.4 changelog. Please add the 0.17.4 section as it appears here. https://github.com/enthought/okonomiyaki/commit/658b66145ff5ceed3887205bfe6f2e95f8352824#diff-d3bb3391c79904494c60ee2ac2f33070

**Comment by kihneman** [comment]
Recreating

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,16 @@
+2019-03-15     1.0.0
+--------------------
```
Comment (itziakos) [inline CHANGELOG]: We are missing a few things that where fixed updated in the 1.0.0 project

**Pull request closed without merging**